src/components/structures/TimelinePanel.js

```diff
@@ -525,25 +526,34 @@ var TimelinePanel = React.createClass({
         //
         if (currentReadUpToEventId && currentReadUpToEventIndex === null &&
                 this._timelineWindow.canPaginate(EventTimeline.FORWARDS)) {
-            return;
+            shouldSendRR = false;
         }
 
         var lastReadEventIndex = this._getLastDisplayedEventIndex({
             ignoreOwn: true
         });
-        if (lastReadEventIndex === null) return;
+        if (lastReadEventIndex === null) {
+            shouldSendRR = false;
+        }
 
         var lastReadEvent = this.state.events[lastReadEventIndex];
+        shouldSendRR = shouldSendRR &&
+            (lastReadEventIndex > currentReadUpToEventIndex &&
+            this.last_rr_sent_event_id != lastReadEvent.getId());
+
+        const shouldSendRM = this.last_rm_sent_event_id != this.state.readMarkerEventId;
 
         // we also remember the last read receipt we sent to avoid spamming the
         // same one at the server repeatedly
-        if ((lastReadEventIndex > currentReadUpToEventIndex &&
-            this.last_rr_sent_event_id != lastReadEvent.getId()) ||
-                this.last_rm_sent_event_id != this.state.readMarkerEventId) {
-
+        if (shouldSendRR || shouldSendRM) {
             this.last_rr_sent_event_id = lastReadEvent.getId();
```

Comment:
Looks like it's possible for lastReadEvent to be undefined here (eg if lastReadEventIndex was null), which would make this blow up.